src/components/PersonInfo.tsx

```diff
@@ -0,0 +1,69 @@
+import { FC } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import cn from 'classnames';
+
+import { Person } from '../types';
+import { findPersonByName } from '../utils/findPersonByName';
+
+interface Props {
+  person: Person;
+  people: Person[];
+}
+
+type ParentLinkProps = {
+  parentName: string | null;
+  parent: { slug: string } | null;
+  className?: string;
+};
+
+const ParentLink: React.FC<ParentLinkProps> = ({
+  parentName,
+  parent,
+  className,
+}) =>
+  parent ? (
+    <Link to={`/people/${parent.slug}`} className={className || ''}>
+      {parentName}
+    </Link>
+  ) : (
+    <>{parentName || '-'}</>
+  );
+
+export const PersonInfo: FC<Props> = ({ person, people }) => {
+  const { personSlug } = useParams<{ personSlug: string }>();
+
+  const mother = findPersonByName(people, person.motherName);
+  const father = findPersonByName(people, person.fatherName);
```

Comment:
The findPersonByName function is used here to find the mother and father objects. Ensure that this function handles cases where the name might be null or undefined to avoid potential runtime errors.